src/main/js/components/TableView.jsx

```diff
@@ -76,12 +111,89 @@ class TableView extends React.Component {
 
     const {currentTableId} = this.state;
     const cols = this.tables.get(currentTableId).columns.models;
-    const view = this.calcVisibilityArray();
+    const view = cols.reduce((a, b) => f.merge({[b.id]: b.visible}, a), {});
     const savedViews = either(localStorage)
       .map(f.prop(["tableViews"]))
       .map(JSON.parse)
       .getOrElse({});
-    localStorage["tableViews"] = JSON.stringify(f.set([currentTableId,name], view, savedViews))
+    localStorage["tableViews"] = JSON.stringify(f.set([currentTableId, name], view, savedViews));
+  };
+
+  cellJumpError = msg => {
+    ActionCreator.showToast(
+      <div id="cell-jump-toast">{msg}</div>,
+      7000
+    )
+  };
+
+  checkGotoCellRequest = (loaded) => {
+    if (!this.pendingCellGoto) {
+      return;
+    }
+
+    const {page} = this.pendingCellGoto;
+    if (loaded >= page || this.tableFullyLoaded) {
+      this.gotoCell(this.pendingCellGoto, loaded);
+    }
+  };
+
+  estimateCellPage = row => 1 + Math.ceil((row - INITIAL_PAGE_SIZE) / PAGE_SIZE);
+
+  gotoCell = ({row, column, page, filter, ignore = false}, nPagesLoaded = 0) => {
+    const columns = this.getCurrentTable().columns.models;
+    if (f.findIndex(f.matchesProperty("id", column), columns) < 0) {
+      this.cellJumpError(i18n.t("table:jump.no_such_column", {col: column}));
+      this.pendingCellGoto = null;
+      return;
+    }
+
+    const cellId = `cell-${this.state.currentTableId}-${column}-${row}`;
+
+    // Helper closure
+    const focusCell = cell => {
+      this.setColumnsVisibility({
+        val: true,
+        coll: [column]
+      });
+      if (filter) {
+        this.changeFilter({
+          filterMode: FilterModes.ID_ONLY,
+          filterValue: row,
+          filterColumnId: "noop",
+          sortColumnId: 0
+        });
+      }
+      const rows = this.getCurrentTable().rows.models;
+      const rowIndex = f.findIndex(f.matchesProperty('id', row), rows);
+      const columns = this.getCurrentTable().columns.models;
```

Comment:
Don't relay on `.models` all the lodash functions should be available one level above.